packages/nextra-theme-docs/src/index.tsx

```diff
@@ -249,11 +239,13 @@ const Content: React.FC<LayoutProps> = ({
       </MenuContext.Provider>
   )
 }
-interface DocsLayoutProps extends PageOpt {
-  meta: Meta
-}
-const createLayout = (opts: DocsLayoutProps, config: DocsThemeConfig) => {
-  const extendedConfig = Object.assign({}, defaultConfig, config, opts)
+
+const createLayout = (opts: PageOpts, config: DocsThemeConfig) => {
+  const extendedConfig = {
+    ...defaultConfig,
+    ...config,
+    unstable_flexsearch: opts.unstable_flexsearch
+  }
```

Comment:
this was a mistake, we don't need to copy all `PageOpts` in `ThemeConfigContext`, but only `unstable_flexsearch`